storm-client/src/jvm/org/apache/storm/utils/LocalState.java

```diff
@@ -123,6 +124,10 @@ private TBase deserialize(ThriftSerializedObject obj, TDeserializer td) {
             } catch (Exception e) {
                 attempts++;
                 if (attempts >= 10) {
+                    if (e.getCause() instanceof TProtocolException) {
+                        LOG.warn("LocalState file is corrupted, resetting state");
```

Comment:
Sorry just one more minor thing. Do you think it makes sense to have the exception plus stack trace here as well? Otherwise it might be hard to debug if this case is hit unexpectedly.

Reply:
Agreed

Reply:
Done! Good idea.